pkg/receive/multitsdb.go

```diff
@@ -777,3 +838,41 @@ func (u *UnRegisterer) MustRegister(cs ...prometheus.Collector) {
 		}
 	}
 }
+
+// Parse, sort, and extend external labels to an initial label set.
+// If an external label shares same name with a label in the initial label set,
+// use the label in the initial label set and inform user about it.
+func parseAndExtendSortedLabels(labelSet labels.Labels, extend []string, logger log.Logger) (labels.Labels, error) {
+	var extendLabels labels.Labels
+	for _, s := range extend {
+		parts := strings.SplitN(s, "=", 2)
+		if len(parts) != 2 {
+			return nil, errors.Errorf("unrecognized label %q", s)
+		}
+		if !model.LabelName.IsValid(model.LabelName(parts[0])) {
+			return nil, errors.Errorf("unsupported format for label %s", s)
+		}
+		extendLabels = append(extendLabels, labels.Label{Name: parts[0], Value: parts[1]})
+	}
+	sort.Sort(extendLabels)
```

Comment:
Currently we display the labels as: sorted receiver's labels, tenant ID label, sorted tenant's external labels.

You think sorting these as a whole would be better? 🤔

Reply:
It would be better to keep labels sorted in case we need to merge them with series labels in certain places. Keeping them unsorted can introduce bugs in different places.

Reply:
Ok I'll do it. I asked because currently I see the thanos main branch keeps sorted receiver's labels - tenant ID label without sorting them as a whole: https://github.com/thanos-io/thanos/blob/main/pkg/receive/multitsdb.go#L566

This is not part of my PR, but maybe here the label set also has to be sorted before returned: https://github.com/thanos-io/thanos/blob/main/pkg/store/labelpb/label.go#L295

Reply:
done

Reply:
> This is not part of my PR, but maybe here the label set also has to be sorted before returned: https://github.com/thanos-io/thanos/blob/main/pkg/store/labelpb/label.go#L295

I haven't done this because there's no confirmation from you.

Tell me if you need labels sorted here too. This doesn't affect my changes. But just 1LOC added.